tests/test_url.py

```diff
@@ -942,10 +942,9 @@ def test_empty_value_for_query():
     assert str(url) == 'http://example.com/path?a='
 
 
-@pytest.mark.xfail
 def test_none_value_for_query():
-    url = URL('http://example.com/path').with_query({'a': None})
-    assert str(url) == 'http://example.com/path?a'
+    with pytest.raises(TypeError):
+        URL('http://example.com/path').with_query({'a': None})
```

Comment:
I changed this one since it looks like we have explicit check against `None` value. So, this is currently the expected behavior.

But, similar to GH-147 (None vs empty for fragment), I would like to add support for `None` value to query values, as well.

What do you think?

Reply:
passing `None` for removing query/fragment entirely looks pretty good for me.
But `None` for dropping the only on query param is cumbersome and counter-intuitive.
Explicit manipulation is better I believe:
```
q = url.query
q = MultiDict(q)  # make it mutable
del q['key']
url.with_query(q)
```